thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsLoader.java

```diff
@@ -0,0 +1,142 @@
+package com.linkedin.thirdeye.rootcause.impl;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.linkedin.thirdeye.anomaly.ThirdEyeAnomalyConfiguration;
+import com.linkedin.thirdeye.anomaly.events.EventType;
+import com.linkedin.thirdeye.api.TimeGranularity;
+import com.linkedin.thirdeye.datalayer.bao.EventManager;
+import com.linkedin.thirdeye.datalayer.dto.EventDTO;
+import com.linkedin.thirdeye.datasource.DAORegistry;
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The type Holiday events loader, which loads the holiday events from Google Calendar periodically.
+ */
+public class HolidayEventsLoader implements Runnable {
+
+  /**
+   * Instantiates a new Holiday events loader.
+   *
+   * @param configuration the configuration
+   */
+  public HolidayEventsLoader(ThirdEyeAnomalyConfiguration configuration) {
+    calendarList = configuration.getCalendars();
+    keyPath = configuration.getRootDir() + configuration.getKeyPath();
+    scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+  }
+
+  /** List of google holiday calendar ids */
+  private static List<String> calendarList;
+  private static String keyPath;
+  private ScheduledExecutorService scheduledExecutorService;
+  private TimeGranularity runFrequency = new TimeGranularity(7, TimeUnit.DAYS);
+
+  private static final Logger LOG = LoggerFactory.getLogger(HolidayEventsLoader.class);
+
+  /** Global instance of the HTTP transport. */
+  private static HttpTransport HTTP_TRANSPORT;
+
+  /** Global instance of the JSON factory. */
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  /** Global instance of the scopes. */
+  private static final Set<String> SCOPES = Collections.singleton(CalendarScopes.CALENDAR_READONLY);
+
+  static {
+    try {
+      HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+    } catch (Exception e) {
+      LOG.error("Can't create http transport with google api.");
```

Comment:
Append the exception at the end of this message:
    LOG.error("Can't create http transport with google api.", e);